### PR TITLE
ipsec: Improve logging

### PR DESCRIPTION
**Pull request by pchaigno**
See commits for details.

```release-note
Fix IPsec error logs to always have all information needed to identify the XFRM configuration on which the error happened.
```

### CODE REVIEW

File: pkg/datapath/linux/ipsec/ipsec_linux.go [hunk 1]
```diff
@@ -67,6 +67,14 @@ const (
 	oldXFRMOutPolicyPriority = 50
 )
 
+type dir string
```
Comment (asauber) [inline pkg/datapath/linux/ipsec/ipsec_linux.go:70]: Given the common meaning, `dir` I think would be better named `direction`.

File: pkg/datapath/linux/node.go [hunk 2]
```diff
@@ -623,13 +624,13 @@ func (n *linuxNodeHandler) encryptNode(newNode *nodeTypes.Node) error {
 			ipsecIPv4Wildcard := &net.IPNet{IP: wildcardIP, Mask: net.IPv4Mask(0, 0, 0, 0)}
 			errs = errors.Join(errs, n.replaceNodeIPSecInRoute(ipsecLocal))
 			spi, err = ipsec.UpsertIPsecEndpoint(ipsecLocal, ipsecIPv4Wildcard, internalIPv4, wildcardIP, 0, ipsec.IPSecDirIn, false)
-			errs = errors.Join(errs, upsertIPsecLog(err, "EncryptNode local IPv4", ipsecLocal, ipsecIPv4Wildcard, spi))
+			errs = errors.Join(errs, upsertIPsecLog(err, "EncryptNode local IPv4", ipsecLocal, ipsecIPv4Wildcard, spi, 0))
```
Comment (asauber) [inline pkg/datapath/linux/node.go:627]: A const declaration for this `0` would be more readable (similar elsewhere in this file).

**Review by asauber — DISMISSED** [review]

File: pkg/datapath/linux/ipsec/ipsec_linux.go (hunk 1 above)
Comment (pchaigno) [reply pkg/datapath/linux/ipsec/ipsec_linux.go:70]: It took me a while to see what you were talking about :sweat_smile: You mean "directory", right? I guess that probably wouldn't be top of mind for someone working on the datapath. I'll change for extra clarity if other changes are requested.

File: pkg/datapath/linux/node.go (hunk 2 above)
Comment (pchaigno) [reply pkg/datapath/linux/node.go:627]: Looks nice, but we don't have this as a common practice across the codebase. It's also somewhat orthogonal to this set of changes (you may have noticed that this zero is the same zero that's on the line above). So I'm not against it, but won't push just for that.

**Review by brb — APPROVED** [review]

**Pull request merged**